From b109b544458eeba65a259926cd73481024486aa2 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Wed, 29 Nov 2017 17:39:05 +0200
Subject: [PATCH 520/706] staging: fsl-dpaa2/mac: defer probe if no mc portal
 is found

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
(cherry picked from commit 021cdd1d1434eee4e22ef794d5fd57ffc64f018d)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index 23a72f1d19bc..028d928d55f6 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -494,8 +494,8 @@ static int dpaa2_mac_probe(struct fsl_mc_device *mc_dev)
 	err = fsl_mc_portal_allocate(mc_dev, FSL_MC_IO_ATOMIC_CONTEXT_PORTAL,
 				     &mc_dev->mc_io);
 	if (err || !mc_dev->mc_io) {
-		dev_err(dev, "fsl_mc_portal_allocate error: %d\n", err);
-		err = -ENODEV;
+		dev_dbg(dev, "fsl_mc_portal_allocate error: %d\n", err);
+		err = -EPROBE_DEFER;
 		goto err_free_netdev;
 	}
 
-- 
2.17.1

